test/MainLangTests.hs

```diff
@@ -0,0 +1,51 @@
+--Tests for the type checker and evaluator in Core/MainLang.hs.
+module MainLangTests where
+
+import           Juvix.Core.MainLang
+
+import           Control.Monad.Except
+import           Numeric.Natural
+import           Prelude
+
+import           Test.Tasty
+import           Test.Tasty.HUnit
+import           Test.Tasty.QuickCheck
+
+import           Test.QuickCheck
+import           Test.QuickCheck.Gen
+
+--property tests of type checker:
```

Comment:
Eventually we'll want to test the core evaluator against the LLVM evaluator, when that's ready, which can be done with random core term generation / fuzzing.

Reply:
Will do in another PR.